aws/data_source_aws_ebs_volume_ids.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"errors"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func dataSourceAwsEbsVolumeIDs() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEbsVolumeIDsRead,
+		Schema: map[string]*schema.Schema{
+			"filter": ec2CustomFiltersSchema(),
+
+			"tags": tagsSchemaComputed(),
+
+			"ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEbsVolumeIDsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	req := &ec2.DescribeVolumesInput{}
+
+	if tags, tagsOk := d.GetOk("tags"); tagsOk {
+		req.Filters = append(req.Filters, buildEC2TagFilterList(
+			keyvaluetags.New(tags.(map[string]interface{})).Ec2Tags(),
+		)...)
+	}
+
+	if filters, filtersOk := d.GetOk("filter"); filtersOk {
+		req.Filters = append(req.Filters, buildEC2CustomFilterList(
+			filters.(*schema.Set),
+		)...)
+	}
+
+	if len(req.Filters) == 0 {
+		req.Filters = nil
+	}
+
+	log.Printf("[DEBUG] DescribeVolumes %s\n", req)
+	resp, err := conn.DescribeVolumes(req)
+	if err != nil {
+		return err
+	}
+
+	if resp == nil || len(resp.Volumes) == 0 {
+		return errors.New("no matching volumes found")
+	}
+
+	volumes := make([]string, 0)
+
+	for _, volume := range resp.Volumes {
+		volumes = append(volumes, *volume.VolumeId)
+	}
+
+	// There is no natural ID to use here, so create a surrogate ID from the volume IDs
+	d.SetId(strconv.Itoa(schema.HashString(strings.Join(volumes, "-"))))
```

Comment:
Nit: Since the resource identifier is meaningless anyway, we can save the complexity/computation and use `resource.UniqueId()` here:

```suggestion
	d.SetId(resource.UniqueId())
```